src/scale_impls/encode.rs

```diff
@@ -180,8 +237,10 @@ fn encode_composite<T>(
     // if the Value provided already ignored all newtype wrappers). If we have nothing
     // to unwrap then ignore this extra encode attempt.
     {
-        let inner_type_id = find_single_entry_with_same_repr(type_id, types);
-        if inner_type_id != type_id {
+        let inner_type_id = find_single_entry_with_same_repr::<R>(type_id, types)
+            .map_err(|err| Error::new(ErrorKind::TypeNotFound(format!("{:?}", err))))?;
+        // Todo/Question: Of course this is completely stupid, we should probably add `PartialEq` bound for R::TypeId in general;
```

Comment:
R::TypeId is typical an u32 or another unsigned integer type?

I don't really understand why it needs to be generic but we need to enforce PartialEq then debug/display impls are not the best to compare different types :P

/cc @jsdw

Reply:
You could avoid any bounds etc and have `find_single_entry_with_same_repr` indicate whether it is about to return a different type ID instead, I think